src/vellum/lib/test_suite.py

```diff
@@ -0,0 +1,175 @@
+import os
+import time
+from typing import Any, Callable, Union
+from uuid import uuid4
+from ..client import Vellum
+from ..types import (
+    ChatMessage,
+    NamedTestCaseJsonVariableValueRequest,
+    NamedTestCaseNumberVariableValueRequest,
+    NamedTestCaseStringVariableValueRequest,
+    SearchResult,
+    TestSuiteRunExecution,
+    TestSuiteRunMetricOutput,
+    TestSuiteRunState,
+    VellumError,
+    TestSuiteRunExternalExecConfigRequest,
+    TestSuiteRunExternalExecConfigDataRequest,
+    ExternalTestCaseExecution,
+)
+
+
+class VellumTestSuiteResults:
+    def __init__(self, id: str) -> None:
+        self.id = id
+        self.client = Vellum(
+            api_key=os.environ.get("VELLUM_API_KEY"),
+        )
+        self.test_suite_run_state = TestSuiteRunState.QUEUED
+        self.test_suite_run_executions: list[TestSuiteRunExecution] | None = None
+
+    def _refresh_test_suite_run_state(self):
+        test_suite_run = self.client.test_suite_runs.retrieve(self.id)
+        self.test_suite_run_state = test_suite_run.state
```

Comment:
Should we just save the whole response (i.e. `_refresh_test_suite_run`) since we're fetching it anyway? Want to avoid a situation where in the future we make extra calls to the API because we fetch in two different places for two different reasons

Reply:
I wanted `_state` to be top level on `self` bc we could then initialize to a non-nullable value and avoid checking nullability everywhere. I'll rename the method to `_refresh_test_suite_run` so that if there are other fields we want to cache, we could do it here